### PR TITLE
Support entt 3.3.0. Contains incompatible changes and a modified API.

### PR DESCRIPTION
**Pull request by GnikDroy**
I didn't find a dev branch so hopefully we can work on this pull request before merging to master. I have written the demo code as well. I will send a PR on that repository as well.

Please feel free to rename the class/function names since I had it customized for my use.

### CODE REVIEW

**Comment by Green-Sky** [comment]
Nice. That was fast.
Now the usual stuff:
- please add yourself to the copyright
- there is a newline missing at the end of the file
- ah yes, the tabs vs spaces war. I'd like them to be tabs :smirk: , sorry

I just noticed the drawback of the new "widget registration", which is that you need to put it in the same namespace (MM). But i guess that's acceptable.

File: imgui_entt_entity_editor.hpp [hunk 1]
```diff
@@ -1,200 +1,156 @@
...
+    }
+
+public:
+    template <class Component>
+    ComponentInfo& registerComponent(const ComponentInfo& component_info)
+    {
+        auto index = entt::type_info<Component>::id();
+        auto [it, insert_result] = component_infos.insert_or_assign(index, component_info);
+        assert(insert_result);
+        auto& [type_index, info] = *it;
```
Comment (Green-Sky) [inline imgui_entt_entity_editor.hpp]: we have an unused variable, how about something like:
```cpp
return std::get<ComponentInfo>(*it);
```

File: imgui_entt_entity_editor.hpp [hunk 2]
```diff
@@ -1,200 +1,156 @@
...
+                }
+
+                if (ImGui::Button("New Entity")) {
+                    e = registry.create();
+                }
+
+                ImGui::Separator();
+
+                if (registry.valid(e)) {
+                    const std::string entity_id = std::to_string(static_cast<int>(registry.entity(e))) + std::to_string(static_cast<int>(registry.version(e)));
```
Comment (Green-Sky) [inline imgui_entt_entity_editor.hpp]: I think `entt::to_integral(e)` would suffice.

Comment (Green-Sky) [reply imgui_entt_entity_editor.hpp]: used to be `entt::to_integer()`

Comment (GnikDroy) [reply imgui_entt_entity_editor.hpp]: Yes. That would indeed be better than the current method.

**Comment by GnikDroy** [comment]
I have made all the requested changes. : )

**Comment by Green-Sky** [comment]
nice idea

**Review by Green-Sky — APPROVED** [review]
so, i had a test run, and it looks like it runs.

**Pull request merged**